src/workbookManager.ts

```diff
@@ -101,14 +107,21 @@ export class WorkbookManager {
         if (!queryProp) {
             throw new Error("No query was found!");
         }
-
+        
+        // Update RefreshOnOpen
         queryProp.parentElement?.setAttribute("refreshOnLoad", refreshOnLoadValue);
+        
+        // Update query details to match queryName
+        dbPr.parentElement?.setAttribute("name", `Query - ${queryName}`);
+        dbPr.parentElement?.setAttribute("description", `Connection to the '${queryName}' query in the workbook.`);
+        dbPr.setAttribute("connection", `Provider=Microsoft.Mashup.OleDb.1;Data Source=$Workbook$;Location=${queryName};`);
+        dbPr.setAttribute("command",`SELECT * FROM [${queryName}]`);
         const connectionId = dbPr.parentElement?.getAttribute("id");
         const newConn = serializer.serializeToString(connectionsDoc);
         zip.file(connectionsXmlPath, newConn);
 
         if (connectionId == "-1" || !connectionId) {
```

Comment:
The connectionId was originally initialized to "-1", so if the connectionId attriubyte is missing we'd got connectionId==-1. We've changed the logic so that connectionId will equal null, so it is no longer necessary and we can use === null instead.